lib/datadog/core/configuration.rb

```diff
@@ -154,9 +152,19 @@ def configuration_for(target, option = nil)
         pin[option] if pin
       end
 
-      def_delegators \
-        :components,
-        :health_metrics
+      def health_metrics
+        HEALTH_METRICS_DEPRECATION_ONLY_ONCE.run do
+          Datadog.logger.warn(
+            'configuration#health_metrics access is deprecated.' \
+                      'Use `Datadog.health_metrics` instead.'
+          )
+        end
+
+        components.health_metrics
+      end
```

Comment:
I.... don't understand this change.

This delegator was setup in the `Datadog::Core::Configuration` module, and this module is included in `::Datadog` and thus this is what allows one to use `Datadog.health_metrics`.

So... I think we're marking as deprecated the way we're telling people to use this...

```
$ bundle exec ruby -e "require 'ddtrace'; Datadog.health_metrics"
W, [2022-05-17T09:30:09.697318 #19971]  WARN -- ddtrace: [ddtrace] configuration#health_metrics access is deprecated.Use `Datadog.health_metrics` instead.
```

...Am I missing something? O_o

Reply:
Argh, I forgot this gets included into `::Datadog`, nvm then.

Reply:
I actually think we should just add these methods directly on `::Datadog` rather than having the module, but lol you already know I think we overuse modules a bit to add behavior within our own codebase :P (other than with integrations, which we don't have many options)